design/baremetal-operator/configmap_for_host_inspection_data.md

```diff
@@ -0,0 +1,148 @@
+to baremetalhost.metal3.io/status annotation so that BMO in the target cluster
+can continue reconciliation from a point it was before pivoting, avoiding redundant
+provisioning.
+
+However, there is a limitation with the current approach, because when inspection
+data is quite big (e.g. when the host has 1000 multipath disks), it is easy to hit
+the global limit of `metadata.annotations` [262144](https://github.com/kubernetes/kubernetes/blob/master/test/integration/controlplane/synthetic_controlplane_test.go#L313) bytes and fail to pivot successfully.
+
+### Goals
+
+Store host inspection data into ConfigMap.
+
+### Non-Goals
+
+None
+
+## Proposal
+
+### User Stories
+
+As an operator, from an ephemeral cluster I would like to pivot/move to a target
+cluster successfully.
+
+## Design Details
+
+Instead of writing hardware details into the status of BareMetalHost, write it
+to a ConfigMap. Baremetalhost and ConfigMap will be parent and child objects
+respectively, and this relationship is controlled via ownerReferences chain. As
+such, the same name & namespace will be used for ConfigMap as its owning object
+BareMetalHost. `metadata.ownerReferences` field in the ConfigMap serves as a
+reference to the owner object BareMetalHost. The data part of ConfigMap will contain
+the same hardware information as currently under `status.hardware` of BareMetalHost.
+
+### Pivoting
+
+[`clusterctl.cluster.x-k8s.io/move-hierarchy`](https://github.com/kubernetes-sigs/cluster-api/blob/0e6428151f2a388b98d1e220a698b85940a5f290/cmd/clusterctl/api/v1alpha3/labels.go#L38) label should be set on BareMetalHost CRD before issuing
+`clusterctl move` so that the entire hierarchy (i.e. including secrets, ConfigMaps)
+is pivoted to the target cluster. Having inspection data in a ConfigMap with
+ownerReferences ensures that the inspection data is available in the target cluster.
+
+### Implementation Details/Notes/Constraints
+
+The life of a ConfigMap is tightly coupled to the corresponding BareMetalHost.
+Creating or deleting a BareMetalHost will result in creation or deletion of a
+ConfigMap respectively.
+
+**Note:** [cross-namespace owner references are not allowed](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications). Thus, ConfigMap and BareMetalHost
+must be in the same namespace.
+
+- BareMetalHost is created
+- Operator creates a ConfigMap with the same name and namespace as BareMetalHost
+    and sets ownerReferences
+- Operator writes name of the ConfigMap to `spec.hardwareInfo` of the BareMetalHost
+- Provisioning requested, operator fills out the ConfigMap with the host
+    inspection data
+- Deprovisioning requested, operator cleans up the inspection data from the
+    ConfigMap
+- BareMetalHost deletion requested, operator deletes the ConfigMap and BareMetalHost
+
+#### BareMetalHost API changes
+
+- `status.hardware` field should be removed ⚠️
```

Comment:
We cannot do this without bumping the BMH CRD version.
We need to get to a point soon where we can do that, but in the meantime it would be good to have a plan for what to do in the meantime (populate both?).